tests/python_package_test/test_engine.py

```diff
@@ -4397,3 +4392,53 @@ def test_quantized_training():
     quant_bst = lgb.train(bst_params, ds, num_boost_round=10)
     quant_rmse = np.sqrt(np.mean((quant_bst.predict(X) - y) ** 2))
     assert quant_rmse < rmse + 6.0
+
+
+@pytest.mark.parametrize("use_weight", [False, True])
+@pytest.mark.parametrize(
+    "test_data",
+    [
+        {
+            "custom_objective": mse_obj,
+            "objective_name": "regression",
+            "df": make_synthetic_regression(),
+            "num_class": 1,
+        },
+        {
+            "custom_objective": multiclass_custom_objective,
+            "objective_name": "multiclass",
+            "df": make_blobs(n_samples=100, centers=[[-4, -4], [4, 4], [-4, 4]], random_state=42),
+            "num_class": 3,
+        },
+    ],
+)
+@pytest.mark.parametrize("num_boost_round", [5, 15])
+@pytest.mark.skipif(getenv("TASK", "") == "cuda", reason="Skip due to ObjectiveFunction not exposed for cuda devices.")
```

Comment:
Why couldn't this also be exposed for the CUDA implementation?

Reply:
Where exactly does it segfault? 🤔 it seems like this _should_ work 😅